test/grub2_test.rb

```diff
@@ -0,0 +1,159 @@
+    allow(::CFA::Grub2::GrubCfg).to receive(:new).and_return(double("GrubCfg").as_null_object)
+    allow(Bootloader::Sections).to receive(:new).and_return(double("Sections").as_null_object)
+    allow(Yast::BootStorage).to receive(:available_swap_partitions).and_return([])
+    allow(Yast::BootStorage).to receive(:gpt_boot_disk?).and_return(false)
+  end
+
+  describe "#read" do
+    before do
+      allow(Yast::BootStorage).to receive(:device_map).and_return(double(empty?: false))
+    end
+
+    it "proposes device map if it is empty" do
+      device_map = double(empty?: true)
+      expect(device_map).to receive(:propose)
+      allow(Yast::BootStorage).to receive(:device_map).and_return(device_map)
+
+      subject.read
+    end
+
+    it "reads bootloader stage1 location" do
+      stage1 = double(Bootloader::Stage1)
+      expect(stage1).to receive(:read)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      subject.read
+    end
+  end
+
+  describe "write" do
+    before do
+      stage1 = double(Bootloader::Stage1, model: double(devices: [], generic_mbr?: false), write: nil)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+      allow(Bootloader::MBRUpdate).to receive(:new).and_return(double(run: nil))
+    end
+
+    it "writes stage1 location" do
+      stage1 = double(Bootloader::Stage1, model: double(devices: [], generic_mbr?: false))
+      expect(stage1).to receive(:write)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      subject.write
+    end
+
+    it "changes pmbr flag as specified in pmbr_action for all boot devices with gpt label" do
+      stage1 = double(Bootloader::Stage1, model: double(devices: ["/dev/sda", "/dev/sdb1"], generic_mbr?: false), write: nil)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      allow(Yast::Storage).to receive(:GetDisk) do |m, dev|
+        case dev
+        when "/dev/sda" then { "device" => "/dev/sda", "label" => "msdos" }
+        when "/dev/sdb1" then { "device" => "/dev/sdb", "label" => "gpt" }
+        else raise "unknown device #{dev}"
+        end
+      end
+
+      expect(subject).to receive(:pmbr_setup).with("/dev/sdb")
+
+      subject.write
+    end
+
+    it "runs grub2-install for all configured stage1 locations" do
+      stage1 = double(Bootloader::Stage1, model: double(devices: ["/dev/sda", "/dev/sdb1"], generic_mbr?: false), write: nil)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      grub2_install = double(Bootloader::GrubInstall)
+      expect(grub2_install).to receive(:execute).with(devices: ["/dev/sda", "/dev/sdb1"])
+      expect(Bootloader::GrubInstall).to receive(:new).with(efi: false).and_return(grub2_install)
+
+      subject.write
+    end
+
+    context "on s390" do
+      before do
+        allow(Yast::Arch).to receive(:architecture).and_return("s390_64")
```

Comment:
Hmmm, so we don't mock the `Yast::Arch` for the default case. That could be fragile.